lib/errors/types.ts

```diff
@@ -0,0 +1,11 @@
+import * as rawErrors from './arsenalErrors';
+
+const entries = Object.keys(rawErrors).map((v) => [v, v]);
+
+/** All possible errors. */
+export type Name = keyof typeof rawErrors
+/** Object containing all errors. It has the format [Name]: "Name" */
+export type Names = { [Name_ in Name]: Name_ };
+
+/** Use types with error.is(types.InternalError) to have nice autocomplete */
+export const types: Names = Object.fromEntries(entries);
```

Comment:
types seems like a bad name here, it feels to me like that would conflict with any other generic type value or import. The best I got right now is just to call it "errors" but that can be redundant when reading it. `error is errors internal error`. You would probably glance over the second error word though. We could also add an extra map without the `Error` suffix in the error names so you could do: `error.is(errors.Internal)`. 

Or maybe just rename `is` to `of` so it reads `error of type internal error`. Feels slightly more grammatically correct. 

Kinda bike shedding on this one though so it is probably fine.

Reply:
That's a good point, `types.ts` was not used anymore, so it is now removed. 
The preferred way to check for an error type is to use directly `error.is.Internal` to avoid some parens and namespaces. It's also more similar to what we already have.